Tasks/Common/ServiceFabricHelpers/Get-ServiceFabricEncryptedText.ps1

```diff
@@ -12,15 +12,30 @@ function Get-ServiceFabricEncryptedText
 
     $defaultCertStoreName = "My"
     $defaultCertStoreLocation = "CurrentUser"
-    $serverCertThumbprint = $ClusterConnectionParameters["ServerCertThumbprint"]
+    $serverCertThumbprints = $ClusterConnectionParameters["ServerCertThumbprint"]
+
+    if ($serverCertThumbprints -is [array])
+    {
+        foreach($serverCertThumbprint in $serverCertThumbprints)
+        {
+            $cert = Get-Item "Cert:\$defaultCertStoreLocation\$defaultCertStoreName\$serverCertThumbprint" -ErrorAction SilentlyContinue
+            if (-not $cert)
+            {
+                break
```

Comment:
does any certificate work for encryption purpose, or only the one (primary/secondary) which has been to connect to cluster? Should we check expiry date also?

Reply:
There is a second set of certificates [Application Certificates](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-cluster-creation-via-arm#application-certificates-optional). Which can be used, but currently we don't support them in the tasks.

Reply:
My question was about server certificates itself. If secondary certificate has been used to connect to cluster, but here we might use primary(when it is first item in $serverCertThumbprints) certificate for encyrption. Will this still work or server will fail in decryption?

Reply:
Yes, primary or secondary will work.